app/customer/customer.py

```diff
@@ -0,0 +1,81 @@
+import datetime
+
+from dataclasses import dataclass
+from typing import Dict, List
+
+from app.customer.car import Car
+from app.customer.chosen_shop import ChosenShop
+from app.shop import Shop
+
+
+@dataclass
+class Customer:
+    name: str
+    home_location: List[int]
+    product_cart: Dict[str, int]
+    money: int
+    car: Car
+    location: List[int] = None
+    chosen_shop: ChosenShop = None
+
+    def pick_shop(self, shops: List[Shop]) -> None:
+        print(f"{self.name} has {self.money} dollars")
+        self.chosen_shop = ChosenShop(self.money)
```

Comment:
Not true. Why do you init by self.money? Better to check inside code:

if self.chosen_shop is None or total_trip_cost <= self.chosen_shop.total_trip_cost: